lib/puppet/network/http/api/v3/indirected_routes.rb

```diff
@@ -50,21 +53,38 @@ def call(request, response)
     return do_exception(response, e)
   end
 
-  def self.url_prefix
+  def self.master_url_prefix
     "#{Puppet[:master_url_prefix]}/v3"
   end
 
+  def self.ca_url_prefix
+    "#{Puppet[:ca_url_prefix]}/v1"
+  end
+
+  def self.url_prefix_for_indirection(indirection_name)
+    IndirectionType.type_for(indirection_name) == :ca ? ca_url_prefix : master_url_prefix
```

Comment:
This is very nitpicky, but... :)

From this line of code it's not necessarily obvious what other values `type_for` might return besides :ca.  If it were changed in the future, this line might no longer be correct, but there wouldn't necessarily be any obvious indicator of that pitfall for the person who was updating the `type_for` method.

It might be better to have a case statement that falls through to a default behavior of throwing an error in case we get an unrecognized value.  Or you could provide a map that contained the mappings from the `type` keyword to the URL string, and then throw an error if `type_for` returns something that we don't have an entry for in the map.

Or, if we're going to try to separate the ca/v1 and master/v3 references so that they're not in the same file, maybe we could move the url mapping into `IndirectionType` as well?  And just add a `url_prefix_for(indirection_name)` method?